### PR TITLE
DOC: initial documentation

### PR DESCRIPTION
**Pull request by ntessore**
## Description

This pull request is to create the initial documentation skeleton for the Cosmology API.

## PR Checklist

- [ ] Give a detailed description of the PR above.
- [ ] Document changes in the `CHANGES.rst` file. See existing changelog for examples.
- [ ] Add tests, if applicable, to ensure code coverage never decreases.
- [ ] Make sure the docs are up to date, if applicable, particularly the docstrings and RST files in `docs` folder.
- [ ] Ensure linear history by rebasing, when requested by the maintainer.

### CODE REVIEW

**Comment by ntessore** [comment]
Docs preview [here](https://cosmology--50.org.readthedocs.build/en/50/). The API sections for components and attributes will be properly filled after #48.

**Comment by ntessore** [comment]
@nstarman I have opted here to use the summary-style tables for properties and methods in the protocols, which is **much** more compact than the full form.

I also decided against creating individual pages for each method, because they contain almost no added value. We should explain the deal with input and output array types in prose in the "how to use" section for users. Other than that, I would argue that no method has a signature that warrants more than the one line summary shown in the table.

**Comment by ntessore** [comment]
@nstarman I managed to hack autosummary to generate something close to what I originally had in mind: a flat list of all methods and properties, and the protocol hierarchy, all linking to the same documentation page with a single source of truth.

The autosummary hack is horrible, but I don't see another way.

**Comment by nstarman** [comment]
The docs look great! In particular, getting the "On this page" to not include the prefix is a major improvement. I'll do a more detailed review and contribute some more docs, probably in the form of docstrings, but a few notes for now:

- the `constants` attribute of [Cosmology](https://cosmology--50.org.readthedocs.build/en/50/api/cosmology.html#cosmology.api.Cosmology) is not linked to [cosmology.api.CosmologyConstantsNamespace](https://cosmology--50.org.readthedocs.build/en/50/api/namespaces.html#cosmology.api.CosmologyConstantsNamespace)
- The base classes of the Components, e.g. [NeutrinoComponent](https://cosmology--50.org.readthedocs.build/en/50/api/components.html#cosmology.api.NeutrinoComponent), don't link to the API pages, e.g. `HasOmegaNu`. Like https://cosmology--50.org.readthedocs.build/en/50/api/attributes.html#cosmology.api.HasTCMB0, these should have grouped sections in the [Attributes](https://cosmology--50.org.readthedocs.build/en/50/api/attributes.html) page.

**Comment by nstarman** [comment]
Needs https://github.com/cosmology-api/cosmology-api/pull/53

**Comment by nstarman** [comment]
Oh no. Rebasing muddled the commit history.

**Comment by ntessore** [comment]
@nstarman Ok, I think that this PR is ready on the technical side, it only needs some more content. Do you want to merge first or do everything in this PR?

**Comment by nstarman** [comment]
Let's merge first! We'll get the docs as solid as possible in quick followups.

File: docs/_ext/sphinx_ext_cosmology_api.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def context_callback(app, name, obj, parent, context):
+    """Callback function to provide extra context for reference."""
+    # add protocols to context if obj is a method or property of Cosmology
+    extra_context = None
+    if parent is cosmo:
+        _, _, membername = name.rpartition(".")
+        protocols = [
+            f"{cls.__qualname__}.{membername}"
```
Comment (nstarman) [inline docs/_ext/sphinx_ext_cosmology_api.py:34]: It looks like this could be made a back-link to the parent Protocol?

Comment (nstarman) [reply docs/_ext/sphinx_ext_cosmology_api.py:34]: For a followup PR.

**Review by nstarman — APPROVED** [review]
Docs look great!

**Review by 89535399501 — COMMENTED** [review]
[61.77199 ](S09543159@gmail.com ) @nstarman

**Pull request merged**